src/instructlab/utils.py

```diff
@@ -438,55 +438,41 @@ def read_taxonomy_file(file_path: str, yaml_rules: Optional[str] = None):
                     parsed_p = p.split(delim)[1]
                     lint_messages.append(parsed_p)
                 logger.error("\n".join(lint_messages))
-                return None, warnings, errors
+                return warnings, errors
 
         validation_errors = validate_yaml(contents, taxonomy_path)
         if validation_errors:
             errors += validation_errors
-            return None, warnings, errors
+            return warnings, errors
+
+        # If the taxonomy file includes a document reference, validate that
+        # we can retrieve the content of the document
+        if "document" in contents:
+            try:
+                _validate_documents(contents["document"])
+            except Exception as e:
+                errors += 1
+                logger.error(f"Failed to load document content for {file_path_p}: {e}")
 
-        # get seed instruction data
-        tax_path = "->".join(taxonomy_path.parent.parts)
-        task_description = contents.get("task_description")
-        documents = contents.get("document")
-        if documents:
-            documents = get_documents(source=documents)
-            logger.debug("Content from git repo fetched")
-
-        for seed_example in contents.get("seed_examples", []):
-            question = seed_example.get("question")
-            answer = seed_example.get("answer")
-            context = seed_example.get("context", "")
-            seed_instruction_data.append(
-                {
-                    "instruction": question,
-                    "input": context,
-                    "output": answer,
-                    "taxonomy_path": tax_path,
-                    "task_description": task_description,
-                    "document": documents,
-                }
-            )
     except Exception as e:
         errors += 1
         raise TaxonomyReadingException(f"Exception {e} raised in {file_path_p}") from e
 
-    return seed_instruction_data, warnings, errors
+    return warnings, errors
 
 
 # TODO: remove `_logger` parameter after instructlab.sdg is fixed.
-def read_taxonomy(_logger, taxonomy, taxonomy_base, yaml_rules):
-    seed_instruction_data = []
+def validate_taxonomy(_logger, taxonomy, taxonomy_base, yaml_rules):
     if os.path.isfile(taxonomy):
-        seed_instruction_data, warnings, errors = read_taxonomy_file(
-            taxonomy, yaml_rules
-        )
+        warnings, errors = validate_taxonomy_file(taxonomy, yaml_rules)
         if warnings:
             logger.warning(
                 f"{warnings} warnings (see above) due to taxonomy file not (fully) usable."
             )
         if errors:
-            raise SystemExit(yaml.YAMLError("Taxonomy file with errors! Exiting."))
+            raise TaxonomyReadingException(
+                yaml.YAMLError("Taxonomy file with errors! Exiting.")
```

Comment:
"Exiting" here is probably wrong now since it's up to the caller to decide what to do with the exception.

Reply:
done